flink-core/src/test/java/org/apache/flink/util/FileUtilsTest.java

```diff
@@ -202,6 +202,24 @@ public void testDeleteDirectoryWhichIsAFile() throws Exception {
 		}
 	}
 
+	@Test
+	public void testDeleteSymbolicLinkDirectory() throws Exception {
+
+		// deleting a symbolic link directory should not delete the files in it.
+
+		File symbolicLinkDirectory = tmp.newFolder();
+		File symbolicLinkDirectoryChild = new File(symbolicLinkDirectory, "child");
```

Comment:
Proposal: `fileInSymbolicLinkDirectory` or `fileInLinkedDirectory` if you choose to change the variable name of the directory.